openwisp_utils/admin_theme/menu.py

```diff
@@ -0,0 +1,240 @@
+        super().__init__(config)
+        name = config.get('name')
+        model = config.get('model')
+        if name:
+            if not isinstance(name, str):
+                raise ImproperlyConfigured(
+                    f'"name" should be a type of "str". Error for config-{config}'
+                )
+            self.name = name
+        else:
+            raise ImproperlyConfigured(f'"name" is missing in the config-{config}')
+        if not model:
+            raise ImproperlyConfigured(f'"model" is missing in config-{config}')
+        if not isinstance(model, str):
+            raise ImproperlyConfigured(
+                f'"model" should be a type of "str". Error for config-{config}'
+            )
+        self.model = model
+        self.label = _(self.get_label(config))
+        self.icon = config.get('icon')
+
+    def get_label(self, config=None):
+        label = config.get('label')
+        if label:
+            return label
+        app_label, model = config['model'].split('.')
+        model_class = registry.apps.get_model(app_label, model)
+        return f'{model_class._meta.verbose_name_plural} {self.name}'
+
+    def create_context(self, request):
+        app_label, model = self.model.split('.')
+        model_label = model.lower()
+        url = reverse(f'admin:{app_label}_{model_label}_{self.name}')
+        view_perm = f'{app_label}.view_{model_label}'
+        change_perm = f'{app_label}.change_{model_label}'
+        user = request.user
+        has_permission_method = (
+            user.has_permission if hasattr(user, 'has_permission') else user.has_perm
+        )
+        if has_permission_method(view_perm) or has_permission_method(change_perm):
+            return {'label': self.label, 'url': url, 'icon': self.icon}
+        return None
+
+    def __eq__(self, other):
+        if self.__class__ != other.__class__:
+            return False
+        return (
+            self.label == other.label
+            and self.model == other.model
+            and self.icon == self.icon
+        )
+
+
+class MenuLink(BaseMenuItem):
+    '''
+    It is used to create any general link by supplying a custom url.
+    Parameters of config: label, url and icon.
+    '''
+
+    def __init__(self, config):
+        super().__init__(config)
+        url = config.get('url')
+        self.set_label(config)
+        if not url:
+            raise ImproperlyConfigured(f'"url" is missing in the config- {config}')
+        if not isinstance(url, str):
+            raise ImproperlyConfigured(
+                f'"url" should be a type of "str". Error for the config- {config}'
+            )
+        self.url = url
+        self.icon = config.get('icon')
+
+    def __eq__(self, other):
+        if self.__class__ != other.__class__:
+            return False
+        return (
+            self.label == other.label
+            and self.url == other.url
+            and self.icon == other.icon
+        )
```

Comment:
nice, what's this for?

Reply:
I added this to compare items during testing of their order. Today I found a way to do the same in a much simpler way so I am removing this code.